AWP-1719: Create a reuseable Text Area component

global-usman opened the pull request:
[AWP-1719](https://team-1637335847501.atlassian.net/browse/AWP-1719)

- Created reusable text area component
- added test cases
- created stories

[AWP-1719]: https://team-1637335847501.atlassian.net/browse/AWP-1719?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

global-usman committed:
AWP-1719/Create a reuseable Text Area component

- Created reusable text area component
- added test cases
- created stories

gm-global commented on src/components/textArea/TextArea.styles.tsx:
```diff
@@ -0,0 +1,112 @@
+`;
+
+export const Heading = styled.div<{ disabled?: boolean }>`
+  color: ${(props) =>
+    props.disabled
+      ? "var(--neutral-200, #9ca3af)"
+      : "var(--primary-300, #115873)"};
```
can we destructure props here and move color property at the end?

gm-global commented on src/components/textArea/TextArea.styles.tsx:
```diff
@@ -0,0 +1,112 @@
+  gap: 8px;
+  align-self: stretch;
+  border-radius: 6px;
+  border: 1px solid
+    ${(props) =>
+      props.isFocused
+        ? "var(--neutral-200, #414141)"
+        : "var(--neutral-200, #9ca3af)"};
+  background: ${(props) =>
+    props.disabled ? "var(--neutral-50, #F9FAFB)" : "var(--system-50, #fff)"};
+  box-shadow: 0px 1px 2px 0px rgba(0, 0, 0, 0.05);
+  flex-shrink: 0;
+  pointer-events: ${(props) => (props.disabled ? "none" : "auto")};
```
Same as above, destructure and move it at the end. Also can we move the dynamic properties in a single method that are using same prop instead of making multiple methods?

gm-global reviewed (approved):
Looks good. Just some nits

global-usman committed:
AWP-1719/create a reuseable text area

- Addressed PR comments

global-usman committed:
Refactor input components to improve style handling

- Renamed `$isFilled` to `isFilled` in `InputProps`
- Updated `StyledInput` to accept `$isFilled` for styled-components
- Adjusted conditional styles in `StyledInput` for error and filled states
- Modified `Input` and `Example` components to use `isFilled` instead of `$isFilled`
- Added `$isFilled` prop to `SearchInput` and updated conditional color styling
